src/intro-to-c/arrays/arrays-intro.md

```diff
@@ -0,0 +1,300 @@
+<summary>Reveal answer</summary>
+
+```
++-----+-----+-----+-----+-----+
+|  0  |  2  |  4  |  6  |  8  |
++-----+-----+-----+-----+-----+
+```
+
+</details>
+
+Let's break down what the array parts of our code does.
+
+```c
+int numbers[5];
+```
+We create an array of integers, with a size of 5. Currently, it is empty.
+
+```c
+numbers[x] = x*2;
+```
+Using our counter variable, x, we access the element in our array at index x.
+Then, we use the assignment operator to set this element equal to what the counter variable is currently, multiplied by 2.
+So, for instance, if x = 3, our loop would essentially be doing this:
+
+```c
+numbers[3] = 3*2;
+```
+Thus, the element at `numbers[3]` would be set to 6.
+Since this loop runs from 0 to 4, then we will apply this "formula" for each cell in the array.
+
+## Being Careful With Arrays
+Arrays are powerful. However, this also means that if you aren't careful,
+you can really mess things up!
+
+### Out of Bounds
+One common mistake one might make is forgetting that arrays in C start at 0, and end at `n - 1`,
+n being the size of the array. If you declare an array with the following statement:
+
+```c
+int arr[5] = {10, 20, 30, 40, 50};
+```
+
+Remember that to access the first element, in this case 10, we would use `arr[0]`. To access the last element,
+we would need to know how large the array is, and subtract 1 from that number. We declared the array with a size of 5,
+so to access the last element in the array, we would use `arr[4]`, which in this case holds 50.
+
+It is **extremely** important to be aware of the sizes of your arrays. If you have an array of size 5, and try accessing
+what is in index 5, (e.g. `arr[5]`) C will let you do this! However, this may cause unintended behavior, and at this point,
+we're crossing into dangerous territory.
+
+```
++------+------+------+------+------+
+|  10  |  20  |  30  |  40  |  50  |   *?*&!(
++------+------+------+------+------+
+                                         ^
+```
+
+Because we accidentally accessed the array element at index 5, we get garbage data! This could even be parts of memory
+that the program is not supposed to be able to access.
+
+Here's an example program to show what can happen if you access
+far beyond the limits of what you defined your array to be:
+
+<details>
+<summary>Show code and output</summary>
+
+```c
+#include <stdio.h>
+
+int main(void) {
+    // define an integer array of size 5, and assign numbers
+    int numbers[5] = {7, 14, 2, 3, 4};
+    // try accessing the 25th index of our numbers array
+    // ...which doesn't make sense, since our array has a size of 5!
+    numbers[25] = 20;
+
+	// try printing every single element
+	for (int i = 0; i <= 25; i++) {
+		// print a warning when i is greater than the last index of our array
+		if (i == 5)
+			printf("\nwarning: entering uncharted territory!\n");
+		// print the number located in the i-th index of numbers
+		printf("i: %d, numbers[i]: %d\n", i, numbers[i]);
```

Comment:
Maybe you want to simplify down some of the output here? Up to you!

```suggestion
		printf("numbers[%d]: %d\n", i, numbers[i]);
```

Reply:
Yea, good idea lol. I updated the output and printf statement